client/src/components/room/createRoom.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright (c) Bryan Hughes <bryan@nebri.us>
+
+This file is part of Home Lights.
+
+Home Lights is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Home Lights is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Home Lights.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import * as React from 'react';
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  TextField
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+export interface CreateRoomDispatch {
+  createRoom: (name: string) => void;
+}
+
+const useStyles = makeStyles({
+  container: {
+    padding: '1em',
+    display: 'flex',
+    'justify-content': 'center',
+    'align-items': 'center'
+  }
+});
+
+export function CreateRoom(props: CreateRoomDispatch): JSX.Element {
+  const [open, setOpen] = React.useState(false);
```

Comment:
This might be clearer when reading top to bottom as `openDialog`

Reply:
Will fix...this was copy-pasted from the material-ui site (as was much of this PR) and probably should have vetted it a bit more